packages/react/src/Link.ts

```diff
@@ -165,7 +165,7 @@ const Link = forwardRef<unknown, InertiaLinkProps>(
     }, prefetchModes)
 
     const regularEvents = {
-      onClick: (event) => {
+      onClick: (event: Parameters<InertiaLinkProps['onClick']>[0]) => {
```

Comment:
This looks a bit ugly but it's basically reusing the type of the first parameter of the `onClick` property defined in `InertiaLinkProps` above